### PR TITLE
Add "export as image" function

### PR DESCRIPTION
**Pull request by fakedestinyck**
However, it seems that this function is only working on Chrome, not available on iOS.
A new module is imported. (`npm install` is required)

### CODE REVIEW

File: src/app/schedule-view/component.ts [hunk 1]
```diff
@@ -142,6 +144,25 @@ export class ScheduleViewComponent implements OnInit, OnDestroy {
     return this.schedules[this.scheduleIndex].statusText;
   }
 
+  public showExport () : void {
+      this.isExporting = true;
+  }
+
+  public downloadImage () : void {
+      var node = document.getElementById('schedule-container');
```
Comment (Bad-Science) [inline src/app/schedule-view/component.ts:152]: The conventional way to get DOM elements in Angular is using the `ViewChild` API. see: https://angular.io/api/core/ViewChild

File: src/app/schedule-view/component.html [hunk 2]
```diff
@@ -8,6 +8,13 @@
         <span class="schedule-switch" (click) = "nextSchedule()">&#9654;</span>
       </div>
       <button class="schedule-button" (click) = "clear()">Clear</button>
+      <!-- <button class="schedule-button">Download ICS</button>
+      <button class="schedule-button" [disabled]="isTemporary">Copy Schedule Link</button> -->
+      <button class="schedule-button" (click) = "showExport()">Export as ...</button>
+      <button *ngIf="isExporting" class="schedule-button" [disabled]="!isExporting" (click) = "downloadImage()">Image</button>
+      <button *ngIf="isExporting" class="schedule-button" [disabled]="!isExporting">iCalendar</button>
```
Comment (Bad-Science) [inline src/app/schedule-view/component.html:15]: Can we remove these since they aren't implemented yet?

File: src/app/schedule-view/component.ts [hunk 3]
```diff
@@ -142,6 +144,25 @@ export class ScheduleViewComponent implements OnInit, OnDestroy {
...
+
+      domtoimage.toPng(node,{bgcolor:"white",quality:1.0})
+        .then(function (dataUrl) {
+            var link = document.createElement('a');
+            link.download = 'mySchedules.png';
+            link.href = dataUrl;
+            link.click();
+        })
+        .catch(function(error) {
+          console.error('oops, something went wrong!', error);
```
Comment (Bad-Science) [inline src/app/schedule-view/component.ts:162]: Could we make this error more specific?

File: package.json [hunk 4]
```diff
@@ -27,6 +27,8 @@
     "@ng-bootstrap/ng-bootstrap": "^1.0.0",
     "classlist.js": "^1.1.20150312",
     "core-js": "^2.5.1",
+    "d3": "^4.10.2",
```
Comment (Bad-Science) [inline package.json:30]: Do we need d3 as a dependency?

**Review by Bad-Science — CHANGES_REQUESTED** [review]
Thanks for the PR!

Requested some changes. Also, from what I remember we discussed either nixing the "Show Export" button or switching to a dropdown instead. Since bootstrap is installed now, can we implement that dropdown? Let me know what you think.

**Comment by fakedestinyck** [comment]
@Bad-Science Sure. I will change it to a dropdown menu

File: src/app/schedule-view/component.ts (hunk 3 above)
Comment (fakedestinyck) [reply src/app/schedule-view/component.ts:162]: @Bad-Science  Actually, specific error description is in error parameter

File: package.json (hunk 4 above)
Comment (fakedestinyck) [reply package.json:30]: Yes. d3 is needed to convert between HTML, canvas, and image. `dom-to-image` package use it as a dependency.

File: src/app/schedule-view/component.html (hunk 2 above)
Comment (fakedestinyck) [reply src/app/schedule-view/component.html:15]: I will.

**Pull request closed without merging**